### PR TITLE
7025 eliminate staging

### LINKED ISSUE #6678: remove reallocate from ZCF
## What is the Problem Being Solved?

#6577 Added a new reallocation API for Zoe, but didn't convert all the uses in existing contracts to the new version.

## Description of the Design

https://github.com/Agoric/agoric-sdk/pull/7136 convered all the uses of `reallocate`, `incrementBy`, and `decrementBy` in agoric-sdk, except for the one in atomicRearrange and tests.

Left to do is :
- remove from dapps
- inform third-parties that the old version is going away
- move `atomicRearrange` onto ZCF
- remove `realllocate` from ZCF

## Security Considerations

The new API is more robust.  Version migration has to be done carefully.

## Test Plan

This ends by deleting the old APIs, which is sufficient test that the methods are gone.

### LINKED ISSUE #7025: eliminate stagings in Vaults Release code
## What is the Problem Being Solved?

The staging mechanism has flaws such as,
- https://github.com/Agoric/agoric-sdk/issues/3850

Thus we need to,
- https://github.com/Agoric/agoric-sdk/issues/6678

For the vaults release we don't have to remove them all, just the ones going on chain.

## Description of the Design

Verify none of these,
- stageDelta
- reallocate
- incrementBy
- decrementBy

are used in these contracts,
- liquidation
- psm
- [provisionPool.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/vats/src/provisionPool.js)
- vaultFactory
- walletFactory
- _anything else in [decentral-test-vaults-config.json](https://github.com/Agoric/agoric-sdk/blob/master/packages/vats/decentral-test-vaults-config.json)_

### PR DESCRIPTION
**Pull request by turadg**
closes: #7025

## Description

This removes all use of zcf staging except in ZCF tests and in `atomicRearrange` that is to replace stagings.

This doesn't close https://github.com/Agoric/agoric-sdk/issues/6678, which should be closed when there is no more `.reallocate` method and the atomicRearrange function is on zcf instead.

### Security Considerations

--
### Scaling Considerations

--
### Documentation Considerations

--
### Testing Considerations

CI

### CODE REVIEW

**Comment by erights** [comment]
> To help advance that, this does remove `.reallocate` from the ZCF type definition so that it's treated as a "private" method of ZCF only for Zoe to use.

Describing such a removal as "private" will confuse ocap programmers in a way that would fail unsafe. It does not deny other users of ZCF access to such "removed" methods. By contrast, for example, changing a Java method from "public" to "private" actually does deny access to those that merely hold a capability to the object. Joe-E, an ocap Java, relies on the soundness of the Java type system to enforce ocap security constraints.

I understand this is "merely" in the PR description. But please revise this text anyway.

File: packages/inter-protocol/src/reserve/doc.md
Comment (erights) [inline packages/inter-protocol/src/reserve/doc.md]: When I "View file" at https://github.com/Agoric/agoric-sdk/blob/7e07e26dce4ef6503eef4af4b4c98c01b3b2af10/packages/inter-protocol/src/reserve/doc.md I see

image

Does this indicate a syntax error?

File: packages/zoe/src/contractFacet/types.js
Comment (erights) [inline packages/zoe/src/contractFacet/types.js]: Can we instead mark it deprecated in a way that vscode will show? If so, I'd prefer that. If it isn't gone yet, it shouldn't seem gone. It shouldn't be so easy to miss.

**Review by erights — APPROVED** [review]
LGTM. Thanks for doing this!!

File: packages/zoe/src/contractFacet/types.js
Comment (turadg) [reply packages/zoe/src/contractFacet/types.js]: The status quo did that. The prior deprecated tag showed the strike though in VSCode. Removing the property from the typedef makes it an error to refer to and thus CI will fail unless it had an error suppression.

I think this is an improvement but I'll wait for your approval about this point. If you disagree and prefer to just have the visual strikethrough and no CI enforcement then I'll restore it.

Comment (erights) [reply packages/zoe/src/contractFacet/types.js]: I do prefer the status quo on that. Due largely to the rest of this PR, it'll be repaired soon anyway when we actually remove it. Thanks.

Comment (turadg) [reply packages/zoe/src/contractFacet/types.js]: Ok, i've removed this commit,
- [chore(types): hide zcf.reallocate](https://github.com/Agoric/agoric-sdk/pull/7136/commits/7e07e26dce4ef6503eef4af4b4c98c01b3b2af10)

File: packages/inter-protocol/src/reserve/doc.md
Comment (turadg) [reply packages/inter-protocol/src/reserve/doc.md]: Yep thanks for catching

**Pull request merged**
